sources/cloudfront/cache_policy.go

```diff
@@ -30,7 +30,7 @@ func cachePolicyListFunc(ctx context.Context, client CloudFrontClient, scope str
 	policies := make([]*types.CachePolicy, 0, len(out.CachePolicyList.Items))
 
 	for i := range out.CachePolicyList.Items {
-		policies = append(policies, out.CachePolicyList.Items[i].CachePolicy)
+		policies[i] = out.CachePolicyList.Items[i].CachePolicy
```

Comment:
no, `policies` gets allocated with len=0 above. This is to have the array pre-allocated, but keep the `append()` pattern for adding items.

Reply:
Oh right, I didn't realise you could do that. Why would you use that pattern over something like the below?

```go
slice := make([]string, len(whatever))
```

Just so you can *always* use append?

Reply:
yes, exactly. There's a certain value in being able to apply a singular pattern (`x=append(x,...)`) everywhere without having to think.